docs/spec_draft.md

```diff
@@ -1444,6 +1445,144 @@ specification. Numeric precision is implementation-defined.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.select_and_scatter
+
+### Semantics
+
+Runs a windowed selection `select` function over `operand` with shape
+`window_dimensions` and stride `window_strides` to produce an amount
+of selected locations whose shape matches `source`. The values in `source`
+are then scattered to the selected locations in output. Multiple scattered
+elements which land in the same output location are combined using the
+`scatter` function.
+
+The following diagram shows how elements in `results` are computed from
+`operand` and `source` using a concrete example.
+
+<p align="center">
+  <img src="spec_draft/select_and_scatter.svg" />
+</p>
+
+Refer to [stablehlo.reduce_window](#stablehloreduce_window) to know more about
+the semantics of `padding`.
+
+The order of reductions is implementation-defined, which means that `scatter`
+and `init_value` must form a monoid to guarantee that the operation produces
+the same results for all inputs on all implementations.
+
+More formally,
+ * Apply [stablehlo.reduce_window](#stablehloreduce_window) with the following
+   parameters to get `selected_values` as output.
+   * `inputs` $=$ { `operand` },
+   * `init_values` $=$ { identity value for `select` },
+   * `window_dimensions`, `window_strides`, and `padding` are used as is,
+   * `body` $=$ `reduce_with_select` defined as
+
+     ```c++
+     reduce_with_select(tensor<E> arg0, tensor<E> arg1) -> tensor<E> {
+      return select(arg0, arg1) ? arg0 : arg1;
+     }
+     ```
+     where `E = element_type(operand)`. `select` is transitive, else the
+     behavior of the operation is undefined.
```

Comment:
After re-reading the following from the spec
```
 the function must hold transitivity (i.e., if select(a, b) and select(b, c) are true, then select(a, c) is also true) so that the selected element does not depend on the order of the elements traversed for a given window.
```

it seems that if the function is not transitive,  then the selection result will become order dependent which anyway is implementation defined.  

Having said that implementation defined seems like a apt treatment. (we need not to say so explicitly as mentioned by your comment)

Reply:
Do we have users for such implementation-defined behavior, i.e. do people rely on that? If not, it would be the best to call this undefined behavior to rule it out completely.

For `reduce`, we couldn't really call that undefined behavior that because several practically-important reducers don't form a monoid with practically-important init_values. But I'd love to do that too if we could.